### ISSUE TITLE
'await-promise' rule is not compatible with AxiosPromise

### ISSUE THREAD

**Issue by stackia** [palantir/tslint#3381]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__: angelozerr/tslint-language-service

#### TypeScript code being linted

```ts
const res = await axios.get(`/users`)
```

with `tslint.json` configuration:

```json
// other rules...
"await-promise": true
```

#### Actual behavior
AxiosPromise is inherited from `Promise`, but tslint gives ERROR `Invalid 'await' of a non-Promise value.`

#### Expected behavior
Types inherited from `Promise` should be recognized as normal promise.
